tools/pubsys/src/aws/publish_ami/mod.rs

```diff
@@ -456,27 +447,30 @@ pub(crate) async fn modify_regional_images(
         let modify_image_future = modify_image(
             user_ids.clone(),
             user_groups.clone(),
-            operation,
+            operation.clone(),
             image_id,
             ec2_client,
-            region,
         );
 
         // Store the region and image ID so we can include it in errors
-        let info_future = ready((region.name().to_string(), image_id.clone()));
+        let info_future = ready((region.clone(), image_id.clone()));
         requests.push(join(info_future, modify_image_future));
     }
 
     // Send requests in parallel and wait for responses, collecting results into a list.
     let request_stream = stream::iter(requests).buffer_unordered(4);
-    let responses: Vec<((String, String), Result<()>)> = request_stream.collect().await;
+    #[allow(clippy::type_complexity)]
+    let responses: Vec<(
+        (String, String),
+        std::result::Result<ModifyImageAttributeOutput, SdkError<ModifyImageAttributeError>>,
+    )> = request_stream.collect().await;
 
     // Count up successes and failures so we can give a clear total in the final error message.
     let mut error_count = 0u16;
     let mut success_count = 0u16;
     for ((region, image_id), modify_image_response) in responses {
         match modify_image_response {
-            Ok(()) => {
+            Ok(_) => {
```

Comment:
Is there any useful data in the response that's being discarded here? As a hypothetical, if it contained the `image_id`, that would let us simplify the tuple type, or ensure that it matched the expected ID.